nautobot/core/jobs/cleanup.py

```diff
@@ -77,10 +99,10 @@ def run(self, *, cleanup_types, max_age=None):
 
             if CleanupTypes.JOB_RESULT in cleanup_types:
                 self.logger.info("Deleting JobResult records prior to %s", cutoff)
-                _, deleted_dict = JobResult.objects.restrict(self.user, "delete").filter(date_done__lt=cutoff).delete()
+                queryset = JobResult.objects.restrict(self.user, "delete").filter(date_done__lt=cutoff)
+                self.recursive_delete_with_cascade(queryset, result)
                 result.setdefault("extras.JobResult", 0)
                 result.setdefault("extras.JobLogEntry", 0)
-                result.update(**deleted_dict)
                 self.logger.info(
                     "Deleted %d JobResult records and their associated %d JobLogEntry records",
                     result["extras.JobResult"],
```

Comment:
Can we make this log message generic to report *all* deleted records? Vaguely like:

```py
for modelname, count in result.items():
    logger.info("As part of deleting %d JobResult records, also deleted %d related %s records",
        deletion_summary["extras.JobResult"], count, modelname)
```